### DISCUSSION TITLE
New installation on an Intel NUC and Debian 12

### DISCUSSION THREAD

**Discussion by sylvarg** [RaspAP/raspap-webgui#1386] [Q&A] [answered]
Hello there,

I have been pretty happy using RaspAP with my Raspberry projects, and it saved me a lot of time when setting up access points, etc. So I was happy when I saw that RaspAP was now working on supporting Debian 12 systems!

I'm now trying to install it on a fresh Debian 12 minimal installation on a small NUC 11 machine, with no success so far. And to be honest, I have absolutely no idea on how to make it work.

Installation is OK (no error messages). But after the initial installation and a reboot, the access point is not visible.
Additionnally, the system displays at boot multiple times:
`[FAILED] Failed to start hostapd.service - Access point and authentIcatIon server for Wi-Fi and Ethernet.`

The wifi card seems correctly recognized, (it's using the iwlwifi driver), while not named `wlan0`
```
root@array1:/# ip a show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute
       valid_lft forever preferred_lft forever
2: eno1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 88:ae:dd:61:a9:ff brd ff:ff:ff:ff:ff:ff
    altname enp2s0
    inet 192.168.2.11/24 brd 192.168.2.255 scope global dynamic noprefixroute eno1
       valid_lft 85936sec preferred_lft 75136sec
    inet6 fe80::f3d1:d046:812b:dc66/64 scope link
       valid_lft forever preferred_lft forever
3: wlp0s20f3: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 78:af:08:7a:b9:4e brd ff:ff:ff:ff:ff:ff
    inet6 fe80::b2a2:c47d:c421:8871/64 scope link tentative
       valid_lft forever preferred_lft forever
```

I have tried modifying `/etc/hostapd/hostapd.conf`,  `/etc/dhcpcd.conf`, `/etc/dnsmasq.d/090_wlan0.conf` and `/etc/raspap/networking/default.json` files to change driver/interface name to match with those in my system, with no success. In addition, I do not know where to look for log files which could help understand what's wrong in my setup.

Any help/advice would be highly appreciated ;)

**Comment by sylvarg** [comment]
Well, things are getting better, but still not completely OK. 
I have modified the name of the interface used in `/etc/hostapd/hostapd.conf`, `/etc/dnsmasq.d/090_wlan0.conf`, `/etc/dhcpcd.conf` and `/etc/raspap/networking/defaults.json`. This time, the hotspot starts correctly, I can connect to it and the connection sharing from `en0` to `wlp0s20f3` is working well.

But the connection to the web interface does not work. When connecting to the ip address 10.3.141.1, the login/password is correctly asked, but a blank page follows.
Note that I had to manually install the `iw` package, since I had errors in lighttpd service mentioning iw was missing. But still the web interface does not work.

Any idea?

EDIT: my error.log from lighttpd:
```
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:PHP Warning:  parse_ini_file(/etc/raspap/hostapd.ini): Failed to open stream: Aucun fichier ou dossier de ce type in /var/www/html/includes/wifi_functions.php on line 160
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:PHP Fatal error:  Uncaught TypeError: floor(): Argument #1 ($num) must be of type int|float, string given in /var/www/html/app/lib/system.php:52
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:Stack trace:
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:#0 /var/www/html/app/lib/system.php(52): floor()
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:#1 /var/www/html/includes/sysstats.php(12): RaspAP\System\Sysinfo->usedMemory()
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:#2 /var/www/html/index.php(41): require_once('...')
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:#3 {main}
2023-09-01 01:36:41: (mod_fastcgi.c.449) FastCGI-stderr:  thrown in /var/www/html/app/lib/system.php on line 52
```

**Reply by billz** [reply]
Hi there, 

The error log was very helpful. The function throwing the error is used to display available RAM on the System page:

https://github.com/RaspAP/raspap-webgui/blob/16a413aa9f47b0d19554edbd0350939481847d43/app/lib/system.php#L49-L53

The php8.2-cgi package is installed on Debian 12, which now properly follows what is known as the [`strict_types` directive](https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.strict). This applies to all math functions, including `floor()` which is used here. php7 will cast values of the wrong type into the expected scalar type if possible.

The fix here is to unambiguously typecast into the expected datatype, like so:

```
return floor(intval($used));
```

We missed this in testing. It's likely there are other cases that need a look. Thanks for raising this.

**Reply by sylvarg** [reply]
Well, happy to see there is solution! 
If the code must be adapted to deal with Debian 12 systems correctly, do not forget to add the installation of the `iw` package in the installer script. I saw this was already dealt with for Ubuntu based systems, but Debian should also be added.
Anyway, thank you very much!

**Reply by billz** [reply, marked as answer]
#1388 should resolve this, thanks!

**Reply by sylvarg** [reply]
Nice, thanks! I'll tray asap. Any idea when it might be merged?

**Reply by billz** [reply]
PR is merged. It will be available in the next general release, probably within the next week.
